Add Infura tutorials (#45)
ConsenSys/doc.quorum-key-manager: docs/Tutorials/ConnectInfura.md

---
description: Connect to an Infura endpoint
---

# Connect to an Infura endpoint

This tutorial walks you through connecting Quorum Key Manager (QKM) to an Infura endpoint.

> This tutorial connects to Infura to allow a QKM-managed account to interact with a smart contract on the Rinkeby network.

Reviewer: This makes it sound like the tutorial is actually connecting to Infura. Maybe something like:
```suggestion
This tutorial demonstrates connecting to Infura to allow a QKM-managed account to interact with a smart contract on the Rinkeby network.
```

    - Port forwarding.
      Run the following commands:

        ```bash
        export POD_NAME=$(kubectl get pods --namespace $QKM-NAMESPACE -l "app.kubernetes.io/name=quorumkeymanager,app.kubernetes.io/instance=quorum-key-manager" -o jsonpath="{.items[0].metadata.name}")

        kubectl --namespace $QKM-NAMESPACE port-forward $POD_NAME 8080:8080
        ```

    - [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/).
      Set up the appropriate Ingress values in the helm chart before deployment in order to have Ingress activated and
      configured according to your solution.

1. Create a new Ethereum account using the `createAccount` method, filling in your own values.
   Change the `localhost` target according to your Ingress configuration and required URL:

    ```bash
    curl --location --request POST 'https://localhost:8080/stores/eth-accounts/ethereum' \
        --header 'Authorization: Basic YWRtaW4tdXNlcg==' \
        --header 'Content-Type: application/json' \
        --data-raw '{
            "keyId": "my-infura-key",
            "tags": {
                "owner": "mySelf"
            }
        }'
    ```

>     The response should yield an Ethereum account address.

Reviewer: ```suggestion
    The response yields an Ethereum account address.
```

1. Send JSON-RPC-based transactions to the Infura node created in step 1 using the Ethereum account created in step 5.
   Change `<MY_ETH_ACCOUNT_ADDRESS>` to the response address from the previous step, and the `localhost` target according
   to your Ingress configuration and required URL:

    ```bash
    curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"eth_sendTransaction","params":[{"from": <MY_ETH_ACCOUNT_ADDRESS>,"to": "0x015C7C7A7D65bbdb117C573007219107BD7486f9","value": "0x1000000"}], "id":1}' http://localhost:8080/nodes/rinkeby-infura
    ```

1. Test that everything worked:

    - Check that Infura has recorded your action with `<YOUR_PROJECT_ID>`.
    - Check that the transaction has been mined.

>       You should be able to view it on Etherscan.

Reviewer: ```suggestion
      You can view it on Etherscan.
```
